otherlibs/systhreads/st_posix.h

```diff
@@ -90,8 +91,14 @@ static void st_thread_join(st_thread_id thr)
 
 static void INLINE st_thread_yield(void)
 {
-#ifndef __linux__
   /* sched_yield() doesn't do what we want in Linux 2.6 and up (PR#2663) */
+  /* but not doing anything here would actually disable preemption (PR#7669) */
+#ifdef HAS_NANOSLEEP
+  struct timespec t;
```

Comment:
This should be `#ifdef __linux__` and it should be put above the two comment lines.  Initially at least we want to restrict the change of implementation to Linux, where the problem is known.  Later you can test MacOS and all the BSDs to see which is better, `nanosleep` or `sched_yield`.